File: shared/remote-config/implementation/src/commonMain/kotlin/com/appotato/shared/remote/config/implementation/NoOpRemoteConfig.kt
package com.appotato.shared.remote.config.implementation

import com.appotato.shared.remote.config.api.RemoteConfig

/**
 * Fallback used when the platform SDK is not configured. Every value falls back to the zero value
 * of its type, which is what callers already have to handle for a key that was never published.
 */
internal class NoOpRemoteConfig : RemoteConfig {
    override fun refresh(onResult: (Boolean) -> Unit): Unit = onResult(false)
    override fun getString(key: String): String = ""
    override fun getBoolean(key: String): Boolean = false
    override fun getLong(key: String): Long = 0L
}

File: iosApp/iosApp/NoOpRemoteConfig.swift
import ComposeApp
import Foundation

/// Mirrors NoOpRemoteConfig in :shared:remote-config:implementation — a missing Firebase config
/// leaves every value at the zero value of its type, it does not stop the app from starting.
final class NoOpRemoteConfig: NSObject, ComposeApp.RemoteConfig {
    func refresh(onResult: @escaping (KotlinBoolean) -> Void) { onResult(KotlinBoolean(bool: false)) }
    func getString(key: String) -> String { "" }
    func getBoolean(key: String) -> Bool { false }
    func getLong(key: String) -> Int64 { 0 }
}
